frontend/js/src/common/brainzplayer/NavidromePlayer.tsx

```diff
@@ -0,0 +1,429 @@
+import * as React from "react";
+import { get as _get, isString, throttle as _throttle } from "lodash";
+import { Link } from "react-router";
+import { faNavidrome } from "../icons/faNavidrome";
+import { DataSourceProps, DataSourceType } from "./BrainzPlayer";
+import {
+  getArtistName,
+  getTrackName,
+  searchForNavidromeTrack,
+} from "../../utils/utils";
+import GlobalAppContext from "../../utils/GlobalAppContext";
+import { dataSourcesInfo } from "../../settings/brainzplayer/BrainzPlayerSettings";
+
+export type NavidromePlayerState = {
+  currentTrack?: NavidromeTrack;
+};
+
+export type NavidromePlayerProps = DataSourceProps;
+
+export default class NavidromePlayer
+  extends React.Component<NavidromePlayerProps, NavidromePlayerState>
+  implements DataSourceType {
+  static contextType = GlobalAppContext;
+  static hasPermissions = (navidromeUser?: NavidromeUser) => {
+    return Boolean(
+      navidromeUser?.md5_auth_token && navidromeUser?.instance_url
+    );
+  };
+
+  static isListenFromThisService(listen: Listen | JSPFTrack): boolean {
+    const musicService = _get(
+      listen,
+      "track_metadata.additional_info.music_service"
+    );
+    return (
+      isString(musicService) && musicService.toLowerCase().includes("navidrome")
+    );
+  }
+
+  public name = "navidrome";
+  public domainName = "navidrome";
+  public icon = faNavidrome;
+  public iconColor = dataSourcesInfo.navidrome.color;
+
+  audioRef: React.RefObject<HTMLAudioElement>;
+  declare context: React.ContextType<typeof GlobalAppContext>;
+
+  debouncedOnTrackEnd: () => void;
+
+  constructor(props: NavidromePlayerProps) {
+    super(props);
+    this.state = {
+      currentTrack: undefined,
+    };
+    this.audioRef = React.createRef();
+
+    this.debouncedOnTrackEnd = _throttle(this.onTrackEnd, 700, {
+      leading: true,
+      trailing: false,
+    });
+  }
+
+  async componentDidMount(): Promise<void> {
+    const { navidromeAuth: navidromeUser = undefined } = this.context;
+    if (NavidromePlayer.hasPermissions(navidromeUser)) {
+      this.setupAudioListeners();
+    }
+    this.updateVolume();
+  }
+
+  componentDidUpdate(prevProps: DataSourceProps) {
+    const { show, volume } = this.props;
+    if (prevProps.show !== show) {
+      if (show) {
+        this.setupAudioListeners();
+      } else {
+        this.pauseAudio();
```

Comment:
Since this function is not used anywhere else, is it needed?
I think you could just call directly:

```suggestion
        this.audioRef.current?.pause()
```